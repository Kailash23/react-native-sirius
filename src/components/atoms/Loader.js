import React from 'react';
import { withTheme } from 'react-native-paper';
import { View, ActivityIndicator, Text } from 'react-native';

function Loader(props) {
  const { theme, loadingText } = props;

  return (
    <View
      style={[
        styles.loader,
        {
          backgroundColor: theme.colors.background
        }
      ]}>
      <ActivityIndicator size="large" color={theme.colors.loader} />
      {loadingText && (
        <Text style={{ color: theme.colors.text }}>{loadingText}</Text>
      )}
    </View>
  );
}
const styles = {
  loader: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    alignContent: 'center'
  }
};

export default withTheme(Loader);
