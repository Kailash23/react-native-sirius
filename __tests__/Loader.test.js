import Loader from '@atoms/Loader';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

configure({ adapter: new Adapter() });

describe('Testing <Loader/>', () => {
  test('Loader should return an activity indicator without text', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });
  test('Loader should return an activity indicator with text', () => {
    const wrapper = shallow(<Loader loadingText={'Test loading text'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
