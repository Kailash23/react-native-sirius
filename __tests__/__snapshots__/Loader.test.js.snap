// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Loader/> Loader should return an activity indicator with text 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  loadingText="Test loading text"
/>
`;

exports[`Testing <Loader/> Loader should return an activity indicator without text 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
/>
`;
